From 9784863870eeee3ca616c0643e42a1e0f8bb1858 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Fri, 17 Aug 2018 17:09:33 +0800
Subject: [PATCH 4426/5242] MLK-19251-02: arm: dts: imx6sx-sabreauto: set
 max7322 O0 for enet IO voltage

commit  5784b8c78c042e3bfe6a78612bbb2726dbf34a73 from
https://source.codeaurora.org/external/imx/linux-imx.git

Due to HW board design, FEC2 depend on FEC1's MII bus. But FEC1's PHY
locates on adapter PHY board that may not connect with sabreauto board.
And FEC1 depend on MAX732X to select 1.8v IO voltage, MAX732X chip is
also installed on adapter board.
In such case, MAX732X expander gpio chip is not registered, and pinctrl
driver cannot detect the gpio chip and return deferred_probe, FEC1 driver
also do defer probe and never has chance to get probed. so its MII bus is
also not regisgered. Then FEC2 also cannot work without MII bus.

To avoid FEC1 depends on MAX732X, move the output IO setting to max732x
driver by setting the default value.

Reviewed-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sabreauto.dts |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6sx-sabreauto.dts b/arch/arm/boot/dts/imx6sx-sabreauto.dts
index 1bc4c1a..8cd0ba0 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto.dts
@@ -273,7 +273,6 @@
 	pinctrl-0 = <&pinctrl_enet1_1>;
 	phy-mode = "rgmii";
 	phy-handle = <&ethphy1>;
-	pinctrl-assert-gpios = <&max7322 0 GPIO_ACTIVE_HIGH>;
 	fsl,magic-packet;
 	status = "okay";
 
@@ -369,6 +368,7 @@
 		reg = <0x68>;
 		gpio-controller;
 		#gpio-cells = <2>;
+		out-default = /bits/ 16 <0x1 0x1>;
 	};
 
 	pmic: pfuze100@08 {
-- 
1.7.9.5

